applications/app/controllers/InteractiveController.scala

```diff
@@ -139,27 +127,53 @@ class InteractiveController(
     }
   }
 
+  def renderDCRAmp(ir: ItemResponse)(implicit request: RequestHeader): Future[Result] = {
+    fItemResponseToFE(Future.successful(ir)) flatMap {
+      case Left((model, blocks)) => {
+        val pageType = PageType.apply(model, request, context)
+        remoteRenderer.getAMPInteractive(wsClient, model, blocks, pageType)
+      }
+      case Right(other) => Future.successful(RenderOtherStatus(other))
+    }
+  }
+
+  def itemResponseToInteractivePickerInputData(ir: ItemResponse)(implicit
+      request: RequestHeader,
+  ): Option[InteractivePickerInputData] = {
+    for {
+      datetime <- ir.content.flatMap(_.webPublicationDate)
+      tags <- ir.content.map(_.tags)
+    } yield InteractivePickerInputData(datetime, tags.toList)
+  }
+
   override def renderItem(path: String)(implicit request: RequestHeader): Future[Result] = {
     val requestFormat = request.getRequestFormat
-    val renderingTier = InteractiveRendering.getRenderingTier(path)
-    (requestFormat, renderingTier) match {
-      case (AmpFormat, USElectionTracker2020AmpPage) => renderInteractivePageUSPresidentialElection2020(path)
-      case (AmpFormat, DotcomRendering)              => renderDCRAmp(path)
-      case (JsonFormat, DotcomRendering)             => renderDCRJson(path)
-      case (HtmlFormat, DotcomRendering)             => renderDCR(path)
-      case _                                         => renderItemLegacy(path)
+    lookupItemResponse(path).flatMap { itemResponse =>
+      itemResponseToInteractivePickerInputData(itemResponse) match {
+        case Some(data) => {
+          val renderingTier = InteractivePicker.getRenderingTier(path, data)
+          (requestFormat, renderingTier) match {
+            case (AmpFormat, USElectionTracker2020AmpPage) => renderInteractivePageUSPresidentialElection2020(path)
+            case (AmpFormat, DotcomRendering)              => renderDCRAmp(itemResponse)
+            case (JsonFormat, _) if request.forceDCR       => renderDCRJson(itemResponse)
+            case (HtmlFormat, DotcomRendering)             => renderDCR(itemResponse)
+            case _                                         => renderItemLegacy(itemResponse)
+          }
+        }
+        case None => Future.successful(Ok("error: 915efe11-2287-45fe-be84-7f9d77d9bad1"))
```

Comment:
Can we include a human readable message here too, as well as the ID?

Reply:
Also, would we really want to `Ok` on this and just return an error string?

Reply:
So... I added a description of the error, mostly that interactive picker input data could not be determined. I also replaced the `Ok` by a `NotFound`.  This is one of those "should never happen" cases, because if the input data was not found it means that the CAPI answer was totally ill-formed, at which point we have a bigger problem anyway.